Vector.c

```diff
@@ -242,6 +261,56 @@ Object* Vector_remove(Vector* this, int idx) {
    }
 }
 
+Object* Vector_softRemove(Vector* this, int idx) {
+   assert(idx >= 0 && idx < this->items);
+
+   Object* removed = this->array[idx];
+   this->array[idx] = NULL;
+
+   this->dirty_count++;
```

Comment:
Miscounting if deleting an index that is already removed.
There should probably be an `assert(removed)` to check for this situation, plus some code to handle this case further down this function.

Reply:
Great catch, fixed.